cpp/include/mcts/UniformNNEvaluationService.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <core/concepts/Game.hpp>
+#include <mcts/NNEvaluation.hpp>
+#include <mcts/NNEvaluationRequest.hpp>
+#include <mcts/NNEvaluationServiceBase.hpp>
+
+template <core::concepts::Game Game>
+class UniformNNEvaluationService : public mcts::NNEvaluationServiceBase<Game> {
+ public:
+  using NNEvaluation = mcts::NNEvaluation<Game>;
+  using NNEvaluationRequest = mcts::NNEvaluationRequest<Game>;
+  using ValueTensor = NNEvaluation::ValueTensor;
+  using PolicyTensor = NNEvaluation::PolicyTensor;
+  using ActionValueTensor = NNEvaluation::ActionValueTensor;
+  using ActionMask = NNEvaluation::ActionMask;
+
+  void evaluate(const NNEvaluationRequest& request) override {
+    ValueTensor value;
+    PolicyTensor policy;
+    ActionValueTensor action_values;
```

Comment:
You can remove this `action_values` declaration here. I'm surprised the compiler isn't complaining about the re-declaration that happens inside the for-loop by default. We should probably enable the gcc setting to complain about this (it's called "shadow" variables).

Also, you can do the `setZero()` calls outside of the for-loop.